handler/payload_subscribe.js

```diff
@@ -1,89 +1,131 @@
-const { buttonPostback, listElement } = require('../lib/facebook');
+const {buttonPostback, listElement} = require('../lib/facebook');
+const subscriptions = require('../lib/subscriptions');
 
 const getHasLabel = function (chat) {
-    return chat.getLabels().then(
-        labels => labelName => labels.indexOf(labelName) !== -1
-    )
-}
+  return chat.getLabels().then(
+    labels => labelName => labels.indexOf(labelName) !== -1
+  )
+};
+
+const disableSubscription = function (psid, timing) {
+  subscriptions.update(psid, timing, false).then(sub => {
+    console.log(`Disabled subscription ${timing} in dynamoDB for ${psid}`);
+
+    if (!sub.morning && !sub.evening) {
+      subscriptions.remove(psid).then(() => {
+        console.log(`Deleted User in dynamoDB with psid ${psid}`);
+      }).catch(error => {
+        console.log(`Deleting user from dynamoDB failed: ${error}`);
+      })
+    }
+  }).catch(error => {
+    console.log(`Updating user from dynamoDB failed: ${error}`);
+  });
+};
+
+const enableSubscription = function (psid, timing) {
+
+  const item = {
+    morning: (timing === 'morning'),
+    evening: (timing === 'evening'),
+  };
+
+  subscriptions.create(psid, item).then(() => {
+    console.log(`Created in dynamoDB ${psid} with ${timing}`);
+  }).catch(error => {
+    console.log('Creating user in dynamoDB failed: ' + error);
+
+    subscriptions.update(psid, timing, true).then(() => {
+      console.log(`Enabled subscription ${timing} in dynamoDB for ${psid}`);
+    }).catch(error => {
+      console.log('Updating user in dynamoDB failed: ' + error);
+    });
+  })
+};
```

Comment:
Warum musst du nach `create` noch mal `update` machen?

Reply:
Nur im Fehlerfall, wenn es die subscription schon gibt

Reply:
Was hälst du davon, wenn wir die Funktion `create` abschaffen und immer `update` machen. Meines Wissens müsste `dynamoDb.put` immer upsert machen

Reply:
Hmm aber `dynamoDb.put` benutzen wir ja gerade in `create`, und das schlägt auf jeden Fall fehl wenn es das Objekt schon gibt. Meinst du `update`?

Reply:
Das schlägt nur fehl, weil da diese Condition ist, dass der Primary Key nicht existieren darf. Wenn man das raus nimmt ist das upsert.

Reply:
OK Die Doku sagt:
PutItem
Creates a new item, or replaces an old item with a new item. If an item that has the same primary key as the new item already exists in the specified table, the new item completely replaces the existing item.

Das ist aber ja eher ungünstig, wir wollen ja nicht das ganze Ding ersetzen

Reply:
Wenn jemand für morgens angemeldet ist und sich auch für abends anmelden will, soll ja seine Anmeldung für morgens nicht gelöscht werden

Reply:
`update` scheint mir da besser zu passen

UpdateItem
Edits an existing item's attributes, or adds a new item to the table if it does not already exist. You can put, delete, or add attribute values. You can also perform a conditional update on an existing item (insert a new attribute name-value pair if it doesn't exist, or replace an existing name-value pair if it has certain expected attribute values).

Kommt mir nur etwas komplizierter vor

Reply:
ah, verstehe. gut. finde zwar bei beiden methoden cooler, wenn man get und dann die entsprechende methode machen würde, aber meinetwegen können wir das erst mal so lassen